### PR TITLE
chore(bench): implement integer casting benchmarks

### PR DESCRIPTION
**Pull request by soonum**
Closes: zama-ai/tfhe-rs-internal#439

### CODE REVIEW

**Comment by soonum** [comment]
@slab-ci integer_full_bench
@slab-ci signed_integer_full_bench

File: tfhe/benches/integer/bench.rs [hunk 1]
```diff
@@ -2146,6 +2158,91 @@ criterion_group!(
...
+    bench_name: &str,
+    display_name: &str,
+    cast_op: F,
+) where
+    F: Fn(&ServerKey, RadixCiphertext, usize),
+{
+    let mut bench_group = c.benchmark_group(bench_name);
+    bench_group
+        .sample_size(15)
+        .measurement_time(std::time::Duration::from_secs(30));
```
Comment (IceTDrinker) [inline tfhe/benches/integer/bench.rs:2176]: aren't we at 10/60s on the other benches ?

File: tfhe/benches/integer/signed_bench.rs [hunk 2]
```diff
@@ -44,16 +44,45 @@ impl Default for ParamsAndNumBlocksIter {
             Err(_) => false,
         };
 
-        if is_multi_bit {
-            let params = vec![PARAM_MULTI_BIT_MESSAGE_2_CARRY_2_GROUP_2_KS_PBS.into()];
+        EnvConfig {
+            is_multi_bit,
+            is_fast_bench,
+        }
```
Comment (IceTDrinker) [inline tfhe/benches/integer/signed_bench.rs:50]: we should find a way to group these things between signed and unsigned, if needed we do a separate benchmarking crate, I think this falls under the issue I tagged you on for the signed/unsigned benchmarks @soonum

**Review by IceTDrinker — COMMENTED** [review]
looks good, I let Thomas check if he's happy

File: tfhe/benches/integer/bench.rs (hunk 1 above)
Comment (soonum) [reply tfhe/benches/integer/bench.rs:2176]: We do a lot of tests for this operations `bit_sizes_tested ** 2`, so I thought that reducing by two these tests would be beneficial in term of execution duration.

File: tfhe/benches/integer/signed_bench.rs (hunk 2 above)
Comment (soonum) [reply tfhe/benches/integer/signed_bench.rs:50]: Agree, I'll to create to integrate that in `benches/utilities.rs`

Comment (IceTDrinker) [reply tfhe/benches/integer/signed_bench.rs:50]: Not needed for this PR, the objective is the new benchmarks, but yeah later/follow up with the issue you were tagged in for benchmarks simplification

File: tfhe/benches/integer/bench.rs (hunk 1 above)
Comment (IceTDrinker) [reply tfhe/benches/integer/bench.rs:2176]: yes indeed

**Review by IceTDrinker — APPROVED** [review]
Looks good to me, to follow up with some simplification or a benchmark crate

File: tfhe/benches/integer/bench.rs [hunk 3]
```diff
@@ -58,27 +55,48 @@ impl Default for ParamsAndNumBlocksIter {
...
-        };
+        } else {
+            vec![8, 16, 32, 40, 64]
+        }
+    } else if env_config.is_fast_bench {
+        FAST_BENCH_BIT_SIZES.to_vec()
+    } else {
+        BENCH_BIT_SIZES.to_vec()
+    }
+}
```
Comment (tmontaigu) [inline tfhe/benches/integer/bench.rs:79]: This should actually be a method of EnvConfig

Comment (IceTDrinker) [reply tfhe/benches/integer/bench.rs:79]: indeed

File: tfhe/benches/integer/signed_bench.rs (hunk 2 above)
Comment (soonum) [reply tfhe/benches/integer/signed_bench.rs]: Done.

**Review by tmontaigu — APPROVED** [review]

**Pull request merged**